src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,105 @@
+package duke;
+
+
+import exceptions.DukeException;
+import tasks.*;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Parser {
+    static final int TODOLENGTH = 4;
+    static final int EVENTLENGTH = 5;
+    static final int DEADLINELENGTH = 8;
+    static final int DELETELENGTH = 6;
+
+    public static void processUserRequest(String task) {
+        if (task.equalsIgnoreCase("list")) {
+            Ui.printTaskList();
+        }
+        else if (task.contains("done")) {
+            TaskList.markTaskDone(task);
+        }
+        else if (task.contains("delete")) {
+            TaskList.deleteTask(task);
+        }
+        else if (task.contains("find")) {
+            Ui.printMatchingTasks(task);
```

Comment:
You can consider declaring the magic strings (done, delete, find, list) with constants. Similar changes can be made in other places as well.